src/NuGet.Core/NuGet.Packaging/Rules/InvalidUndottedFrameworkRule.cs

```diff
@@ -227,8 +227,9 @@ internal static IEnumerable<PackagingLogMessage> ValidateFiles(IEnumerable<strin
                 {
                     foreach (ContentItem item in group.Items)
                     {
-                        var frameworkString = (string)item.Properties["tfm_raw"];
-                        if (string.IsNullOrEmpty(frameworkString))
+                        var exists = item.Properties.TryGetValue("tfm_raw", out var frameworkRaw);
+                        string frameworkString = (string)frameworkRaw;
+                        if (!exists || !string.IsNullOrEmpty(frameworkString))
```

Comment:
nit: 

Just for reference...stylistically, the below might work.
```cs
if(!item.Properties.TryGetValue("tfm_raw", out string frameworkRaw))
{
}

```cs

Reply:
that doesn't work. It needs to be explicitly cast :(